### PR TITLE
Test migration path generation

### PR DESCRIPTION
**Pull request by ajcvickers**
Part of #32738

There are no functional product changes here, Instead this PR adds code to allow migrations to be generated without actually creating the files. This allows testing of the current behavior such that we can make changes and have confidence that we know what the impact is.

### CODE REVIEW

File: src/EFCore.Design/Migrations/Design/IMigrationsScaffolder.cs [hunk 1]
```diff
@@ -19,12 +19,14 @@ public interface IMigrationsScaffolder
     /// <param name="rootNamespace">The project's root namespace.</param>
     /// <param name="subNamespace">The migration's sub-namespace.</param>
     /// <param name="language">The project's language.</param>
+    /// <param name="dryRun">If true, then nothing is actually written to disk.</param>
```
Comment (AndriySvyryd) [inline src/EFCore.Design/Migrations/Design/IMigrationsScaffolder.cs:22]:
```suggestion
    /// <param name="dryRun">If <see langword="true" />, then nothing is actually written to disk.</param>
```
And elsewhere

File: src/EFCore.Design/Migrations/Design/IMigrationsScaffolder.cs [hunk 2]
```diff
@@ -19,12 +19,14 @@ public interface IMigrationsScaffolder
...
     /// <returns>The scaffolded migration.</returns>
     ScaffoldedMigration ScaffoldMigration(
         string migrationName,
         string? rootNamespace,
         string? subNamespace = null,
-        string? language = null);
+        string? language = null,
+        bool dryRun = false);
```
Comment (AndriySvyryd) [inline src/EFCore.Design/Migrations/Design/IMigrationsScaffolder.cs:29]: Make it default to false in other methods as well.

File: src/EFCore.Design/Migrations/Design/MigrationsScaffolder.cs [hunk 3]
```diff
@@ -233,13 +236,15 @@ public virtual MigrationFiles RemoveMigration(string projectDir, string rootName
     /// <param name="rootNamespace">The project's root namespace.</param>
     /// <param name="force">Don't check to see if the migration has been applied to the database.</param>
     /// <param name="language">The project's language.</param>
+    /// <param name="dryRun">If true, then nothing is actually written to disk.</param>
     /// <returns>The removed migration files.</returns>
     // TODO: DRY (file names)
```
Comment (AndriySvyryd) [inline src/EFCore.Design/Migrations/Design/MigrationsScaffolder.cs:241]: File an issue or remove

File: src/EFCore.Design/Design/OperationExecutor.cs [hunk 4]
```diff
@@ -370,14 +379,15 @@ public class RemoveMigration : OperationBase
 
             var contextType = (string?)args["contextType"];
             var force = (bool)args["force"]!;
+            var dryRun = (bool?)args["dryRun"]!;
 
-            Execute(() => executor.RemoveMigrationImpl(contextType, force));
+            Execute(() => executor.RemoveMigrationImpl(contextType, force, dryRun == true));
```
Comment (AndriySvyryd) [inline src/EFCore.Design/Design/OperationExecutor.cs:384]: Are these changes needed?

**Review by AndriySvyryd — APPROVED** [review]

File: src/EFCore.Design/Design/OperationExecutor.cs (hunk 4 above)
Comment (ajcvickers) [reply src/EFCore.Design/Design/OperationExecutor.cs:384]: Technically no, since we don't have any tests that run this. But we should have these tests, and we will need it then, like we do for `AddMigration` now.
